ucp/_libs/ucp_py.pyx

```diff
@@ -729,7 +729,18 @@ def get_endpoint(peer_ip, peer_port):
     global reader_added
 
     ep = Endpoint()
-    ep.connect(peer_ip, peer_port)
+    deadline = time.time() + timeout
```

Comment:
`timeout=None` is a common convention for "try forever".  You might consider adding support for it.